sklearn/calibration.py

```diff
@@ -25,6 +27,368 @@
+            base_estimator.decision_function will be used first and if not
+            available base_estimator.predict_proba
+
+    beta : float in [0, 1], optional (default=None)
+        beta value to be used in case strategy == 'f_beta'
+
+    threshold : float in [0, 1] or None, (default=None)
+        In case strategy is 'max_tpr' or 'max_tnr' this parameter must be set
+        to specify the threshold for the true negative rate or true positive
+        rate respectively that needs to be achieved
+
+    pos_label : object, optional (default=1)
+        Object representing the positive label
+
+    cv : int, cross-validation generator, iterable or 'prefit', optional
+        (default=3). Determines the cross-validation splitting strategy.
+        If cv='prefit' the base estimator is assumed to be fitted and all data
+        will be used for the calibration of the probability threshold
+
+    Attributes
+    ----------
+    decision_threshold_ : float
+        Decision threshold for the positive class. Determines the output of
+        predict
+
+    std_ : float
+        Standard deviation of the obtained decision thresholds for when the
+        provided base estimator is not pre-trained and the decision_threshold_
+        is computed as the mean of the decision threshold of each
+        cross-validation iteration. If the base estimator is pre-trained then
+        std_ = None
+
+    classes_ : array, shape (n_classes)
+        The class labels.
+
+    References
+    ----------
+    .. [1] Receiver-operating characteristic (ROC) plots: a fundamental
+           evaluation tool in clinical medicine, MH Zweig, G Campbell -
+           Clinical chemistry, 1993
+
+    """
+    def __init__(self, base_estimator, strategy='roc', method=None, beta=None,
+                 threshold=None, pos_label=1, cv=3):
+        self.base_estimator = base_estimator
+        self.strategy = strategy
+        self.method = method
+        self.beta = beta
+        self.threshold = threshold
+        self.pos_label = pos_label
+        self.cv = cv
+
+    def fit(self, X, y):
+        """Fit model
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data
+
+        y : array-like, shape (n_samples,)
+            Target values. There must be two 2 distinct values
+
+        Returns
+        -------
+        self : object
+            Instance of self
+        """
+        if (not hasattr(self.base_estimator, 'decision_function') and
+                not hasattr(self.base_estimator, 'predict_proba')):
+            raise TypeError('The base_estimator needs to implement either a '
+                            'decision_function or a predict_proba method')
+
+        if self.strategy not in ('roc', 'f_beta', 'max_tpr', 'max_tnr'):
+            raise ValueError('strategy can either be "roc" or "max_tpr" or '
+                             '"max_tnr. Got {} instead'.format(self.strategy))
```

Comment:
typo: "max_tnr" (second " missing)

Reply:
thnx for taking the time, will fix those; There are also a few review points from @jnothman that I haven't gone over yet because I am waiting for the general verdict irt how this feature will be finalised before applying new changes.